pkg/server/delta/v3/watches.go

```diff
@@ -0,0 +1,50 @@
+package delta
+
+import (
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/server/stream/v3"
+)
+
+// watches for all delta xDS resource types
+type watches struct {
+	deltaResponses    map[string]watch
+	deltaTerminations map[string]chan struct{}
+
+	// Organize stream state by resource type
+	deltaStreamStates map[string]stream.StreamState
+
+	// Opaque resources share a muxed channel. Nonces and watch cancellations are indexed by type URL.
+	deltaMuxedResponses chan cache.DeltaResponse
+}
+
+// watch contains the necessary modifiables for receiving resource responses
+type watch struct {
+	responses chan cache.DeltaResponse
+	cancel    func()
+	nonce     string
+}
+
+// newWatches creates and initializes watches.
+func newWatches() watches {
+	// deltaMuxedResponses needs a buffer to release go-routines populating it
```

Comment:
Seems like several of these use a buffer, update comment?

Reply:
Well the only thing that uses a buffer would be` deltaMuxesResponses` since that's the only thing that makes a `chan`.
```go
// watches for all delta xDS resource types
type watches struct {
	deltaResponses    map[string]watch
	deltaTerminations map[string]chan struct{}

	// Organize stream state by resource type
	deltaStreamStates map[string]stream.StreamState

	// Opaque resources share a muxed channe
	deltaMuxedResponses chan cache.DeltaResponse
}
```
All the other types are just maps

Reply:
Oh I misread the size arguments as buffers. Maybe move the comment closer to the channel declaration?

Looking at this I wonder if we can use a single map instead of having three different maps keying of a type URL, but we can defer that to another PR if you'd prefer